STM32/USBprint/Src/USBprint.c

```diff
@@ -22,13 +23,23 @@ int USBnprintf(const char * format, ... )
     len += vsnprintf(&buffer[len], sizeof(buffer) - len, format, args);
     va_end (args);
 
-    usb_cdc_transmit((uint8_t*)buffer, len);
-    return len;
+    /* Error code captured in lower level module */
+    ssize_t ret = usb_cdc_transmit((uint8_t*)buffer, len);
+    if(ret >= 2) {
+        return ret - 2;
+    }
+    else if(ret >= 0) {
+        return 0;
+    }
+    else {
+        return ret;
+    }
 }
 
 ssize_t writeUSB(const void *buf, size_t count)
 {
-    return usb_cdc_transmit(buf, count);
+    /* Error code captured in lower level module */
+    return usb_cdc_transmit((const uint8_t*)buf, count);
```

Comment:
Is there a reason why the input is cast to (const uint8_t*) rather than (uint8_t *) in USBnprintf?

Reply:
Error